### PR TITLE
Support GKD and on-policy distillation

### PR DESCRIPTION
**Pull request by tastelikefeet**
# PR type
- [ ] Bug Fix
- [x] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information

Write the detail information belongs to this PR.

## Experiment results

Paste your experiment result here(if needed).

### CODE REVIEW

File: src/twinkle/loss/gkd.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            end = min(start + chunk_size, num_valid_int)
+            s_chunk = student_logits[start:end]
+            t_chunk = teacher_logits[start:end]
+
+            s_log_probs = F.log_softmax(s_chunk, dim=-1)
+            t_log_probs = F.log_softmax(t_chunk, dim=-1)
+            del s_chunk, t_chunk
+
+            if beta == 0:
+                # Forward KL: KL(S || T)
```
Comment (hjh0119) [inline src/twinkle/loss/gkd.py]: Forward KL: KL(T || S)
